wandb/apis/public.py

```diff
@@ -2141,39 +2152,27 @@ def updated_at(self):
 
     @property
     def description(self):
-        return self._attrs["description"]
+        return self._description
 
     @description.setter
     def description(self, desc):
-        self._attrs["description"] = desc
+        self._description = desc
 
     @property
     def type(self):
         return self._attrs["artifactType"]["name"]
 
     @property
     def name(self):
-        """Stable name you can use to fetch this artifact."""
-        # TODO: All this logic should move to the backend.
-        if ":" not in self.artifact_name:
-            # this is a digest lookup
-            return self.artifact_name
-        artifact_collection_name = self.artifact_name.split(':')[0]
-        for alias in self._attrs["aliases"]:
-            if alias["artifactCollectionName"] == artifact_collection_name and re.match(r"^v\d+$", alias["alias"]):
-                return '%s:%s' % (artifact_collection_name, alias["alias"])
-        
-        raise ValueError('Unexpected API result.')
+        return '%s:v%s' % (self._sequence_name, self._version_index)
 
     @property
     def aliases(self):
-        if ":" not in self.artifact_name:
-            # this is a digest lookup
-            return []
-        artifact_collection_name = self.artifact_name.split(':')[0]
-        return [a["alias"] for a in self._attrs["aliases"]
-            if not re.match(r"^v\d+$", a["alias"]) and
-                a["artifactCollectionName"] == artifact_collection_name]
+        return self._aliases
+
+    @aliases.setter
+    def aliases(self, aliases):
+        self._aliases = aliases
```

Comment:
We might want to validate the aliases don't have colons or other invalid characters.

Reply:
Gorilla validates this, but we'll get a better error message if we validate it here. Will fix.

Reply:
Realized that gorilla allows aliases with colons in them...whoops. Fixing.

Reply:
Fixed!